### PR TITLE
Added support for Supabase Storage

### PR DESCRIPTION
**Pull request by jawngee**
## Description

- [x] Adds support for using Supabase Storage as a storage location.

## Type of Change

- [ ] Bugfix
- [x] Improvement
- [ ] New Feature
- [ ] Refactor / codestyle updates
- [ ] Other, please describe:

## Requirements Checklist

- [ ] New / updated tests are included
- [x] All tests are passing locally
- [x] Performed a self-review of the submitted code

If adding a new feature:

- [ ] Documentation was added/updated. PR:

### CODE REVIEW

**Comment by jawngee** [comment]
Added fixes for linting and tests.

File: api/package.json [hunk 1]
```diff
@@ -81,7 +81,8 @@
 		"@directus/drive": "workspace:*",
 		"@directus/drive-azure": "workspace:*",
 		"@directus/drive-gcs": "workspace:*",
-		"@directus/drive-s3": "workspace:*",
+    "@directus/drive-s3": "workspace:*",
+    "@directus/drive-supabase": "workspace:*",
```
Comment (rijkvanzanten) [inline api/package.json:85]: Indentation inconsistency

File: api/src/middleware/respond.ts [hunk 2]
```diff
@@ -75,7 +75,15 @@ export const respond: RequestHandler = asyncHandler(async (req, res) => {
...
-			return res.status(200).send(exportService.transform(res.locals.payload?.data, 'csv'));
+			return res
+				.status(200)
+				.send(
+					exportService.transform(
+						res.locals.payload?.data,
+						'csv',
+						req.query?.separator ? { delimiter: req.query.separator as string } : {}
+					)
+				);
```
Comment (rijkvanzanten) [inline api/src/middleware/respond.ts:86]: Lets do any non-supabase related features / enhancement in separate PRs 👍🏻

**Review by rijkvanzanten — CHANGES_REQUESTED** [review]
Thanks @jawngee! 

Lets move the CSV delimiter changes to a separate PR to not conflate the features here. Other than that, this is really exciting 🙂

**Comment by rijkvanzanten** [comment]
Closing this as the PR became stale. #16830 is a great follow up, we're tracking final implementation in #16903

**Pull request closed without merging**